GoogleDataTransport/GDTLibrary/GDTStorage.m

```diff
@@ -111,7 +111,16 @@ - (void)storeEvent:(GDTEvent *)event {
 
     // If running in the background, save state to disk and end the associated background task.
     if (bgID != GDTBackgroundIdentifierInvalid) {
-      [NSKeyedArchiver archiveRootObject:self toFile:[GDTStorage archivePath]];
+      if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
+        NSData *data = [NSKeyedArchiver archivedDataWithRootObject:self
+                                             requiringSecureCoding:YES
+                                                             error:nil];
+        [data writeToFile:[GDTStorage archivePath] atomically:YES];
+      } else {
+#if !defined(TARGET_OS_MACCATALYST)
```

Comment:
is it possible for TARGET_OS_MACCATALYST to ever be defined when !@available(macOS 10.13)?

Reply:
Yep. It's not executed, but it still tries to compile and fails when building for Catalyst.